src/mscorlib/shared/System/Globalization/DateTimeParse.cs

```diff
@@ -4629,9 +4632,15 @@ private static Exception GetDateTimeParseException(ref DateTimeResult result)
                 case ParseFailureKind.FormatWithParameter:
                     return new FormatException(SR.Format(SR.GetResourceString(result.failureMessageID), result.failureMessageFormatArgument));
                 case ParseFailureKind.FormatBadDateTimeCalendar:
-                    return new FormatException(SR.Format(SR.GetResourceString(result.failureMessageID), result.calendar));
+                    return new FormatException(SR.Format(SR.GetResourceString(result.failureMessageID), new string(result.originalDateTimeString), result.calendar));
+                case ParseFailureKind.FormatWithOriginalDateTime:
+                    return new FormatException(SR.Format(SR.GetResourceString(result.failureMessageID), new string(result.originalDateTimeString)));
+                case ParseFailureKind.FormatWithFormatSpecifier:
+                    return new FormatException(SR.Format(SR.GetResourceString(result.failureMessageID), new string(result.failedFormatSpecifier)));
+                case ParseFailureKind.FormatWithOriginalDateTimeAndParameter:
+                    return new FormatException(SR.Format(SR.GetResourceString(result.failureMessageID), new string(result.originalDateTimeString), result.failureMessageFormatArgument));
                 default:
-                    Debug.Fail("Unkown DateTimeParseFailure: " + result);
+                    Debug.Fail("Unkown DateTimeParseFailure: " + nameof(DateTimeResult));
```

Comment:
Can you fix the typo?

Reply:
Also, `nameof(DateTimeResult)` will just give the string `DateTimeResult`. I think you mean `result.failure.ToString()` ?